### PR TITLE
Feature #576: Added basic support for iframe embedding

### PR DESCRIPTION
**Pull request by FloHin**
- add parameter handler for embedded, controls, header and zoomControls
- add option to set zoom via parameter
- add feature to display compact zoomControl widget

### CODE REVIEW

File: demo/embedded_demo.html [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            <td>"0" = OFF</td>
+            <td>Activates the embedded mode: controls are hidden (unless other params are given) and sheet will be
+                rendered more compactly
+            </td>
+        </tr>
+        <tr>
+            <td>showControls</td>
+            <td>"0" and "1"</td>
+            <td>"0" = OFF</td>
+            <td>If active, the controls of OSMD are shown like usually.</td>
```
Comment (sschmidTU) [inline demo/embedded_demo.html]: some controls like zoom level are shown, not all OSMD controls like options, re-render and so on.

File: demo/embedded_demo.html [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        </tr>
+        <tr>
+            <td>showZoomControl</td>
+            <td>"0" and "1"</td>
+            <td>"0" = OFF</td>
+            <td>If showControls == 0 and showZoomControl == 1, a small zoomControl widget will be display to safe
```
Comment (sschmidTU) [inline demo/embedded_demo.html]: displayed

File: demo/embedded_demo.html [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            <td>showHeader</td>
+            <td>"0" and "1"</td>
+            <td>"0" = OFF</td>
+            <td>If activated, the OSMD title will be rendered on top of the sheet</td>
+        </tr>
+        <tr>
+            <td>zoom</td>
+            <td>floating number between 0.1 and 5.0</td>
+            <td>1.0</td>
+            <td>If given, the zoom will be set at loading time</td>
```
Comment (sschmidTU) [inline demo/embedded_demo.html]: zoom -> OSMD zoom level. no biggie, but it could be unclear what kind of zoom is used

File: demo/index.js [hunk 4]
```diff
@@ -48,7 +48,9 @@ import { OpenSheetMusicDisplay } from '../src/OpenSheetMusicDisplay/OpenSheetMus
     },
 
     zoom = 1.0,
-    // HTML Elements in the page
```
Comment (sschmidTU) [inline demo/index.js]: why remove the comment?

File: demo/index.js [hunk 5]
```diff
@@ -256,14 +329,15 @@ import { OpenSheetMusicDisplay } from '../src/OpenSheetMusicDisplay/OpenSheetMus
         if (!isCustom) {
             str = sampleFolder + selectSample.value;
         }
-        zoom = 1.0;
+        // zoom = 1.0;
```
Comment (sschmidTU) [inline demo/index.js]: awesome. not resetting the zoom level between sample changes is more correct anyways. if anything, we need a reset zoom button in the (main) demo.

**Comment by sschmidTU** [comment]
Looks great! I just found a few minor points, see comments.
When the user hasn't started the local server for OSMD, should we show some kind of notice? Maybe in the div or console?
image
We could say something like `OSMD was expected to run at localhost:[port], could not connect`.
Also, maybe we could put that information somewhere in the embedded demo.html.

**Comment by sschmidTU** [comment]
this closes #576

File: demo/embedded_demo.html (hunk 1 above)
Comment (matt-uib) [reply demo/embedded_demo.html]: we need to define now what controls and other stuff we want to allow the embedder to customize/hide/show.
I would say only developers would need something like re-render which should not be part of the embedding options.

Comment (matt-uib) [reply demo/embedded_demo.html]: But on this demo page should change the text "the controls of OSMD are shown like usually" to be more correct.
Also the "Default" values given don't seem to be correct, right? At least the showZoomControl seems to be on by default, but off is written there.

File: demo/embedded_demo.html [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+            <td>"auto"</td>
+            <td>If given, this attribute will refresh the state of the css attribute overflow to control the
+                scrollbar.
+            </td>
+        </tr>
+    </table>
+</div>
+<div>
+    <center>
+        <iframe src="http://localhost:8000/?embedded&showZoomControl=1&zoom=0.5" width="800" height="1500"
```
Comment (matt-uib) [inline demo/embedded_demo.html]: Frame-border should be possible to be switched off.
Actually we might need: thickness and color of border.
Nice would be also sheet background color.

Comment (FloHin) [reply demo/embedded_demo.html]: This has to be defined by the window which displays the iframe! Not inside, but from the outside

File: demo/index.js (hunk 4 above)
Comment (FloHin) [reply demo/index.js]: Added it again

**Pull request merged**